src/__tests__/key_revocation.ts

```diff
@@ -0,0 +1,95 @@
+/**
+ * @jest-environment ./build/index.js
+ */
+
+import {CeramicApi} from "@ceramicnetwork/common";
+import * as sha256 from "@stablelib/sha256";
+import * as uint8arrays from "uint8arrays";
+import KeyDidResolver from "key-did-resolver";
+import ThreeIdResolver from "@ceramicnetwork/3id-did-resolver";
+import { Resolver } from "did-resolver";
+import {DID, GeneralJWS} from "dids";
+import ThreeIdProvider from "3id-did-provider";
+import {TileDocument} from "@ceramicnetwork/stream-tile";
+import {waitForAnchor} from "../utils";
+import * as didJWT from "did-jwt";
+
+declare global {
+    const ceramic: CeramicApi;
+    const ceramicClient: CeramicApi;
+}
+
+async function extractKid(did: DID): Promise<string> {
+    const jws = await did.createJWS(`KID ${Math.random()}`);
+    const headerRaw = jws.signatures[0].protected;
+    const header = JSON.parse(
+        uint8arrays.toString(uint8arrays.fromString(headerRaw, "base64url"))
+    );
+    return header.kid as string;
+}
+
+function toGeneralJWS(jws: string): GeneralJWS {
+    const [protectedHeader, payload, signature] = jws.split('.')
+    return {
+        payload,
+        signatures: [{ protected: protectedHeader, signature }],
+    }
+}
+
+test('key revocation', async () => {
+    // 1. Setup initial keys
+    const seedString = `first-seed-${Math.random()}`
+    const seed = sha256.hash(uint8arrays.fromString(seedString));
+    const keyDidResolver = KeyDidResolver.getResolver();
+    const threeIdResolver = ThreeIdResolver.getResolver(ceramic);
+    const resolver = new Resolver({
+        ...threeIdResolver,
+        ...keyDidResolver,
+    });
+    const threeIdProvider = await ThreeIdProvider.create({
+        getPermission: async () => [],
+        authSecret: seed,
+        authId: "first",
+        ceramic: ceramicClient,
+    });
+    const did = new DID({
+        provider: threeIdProvider.getDidProvider(),
+        resolver: resolver,
+    });
+    await did.authenticate();
+    ceramicClient.did = did;
+    const firstKid = await extractKid(ceramic.did);
+    const firstSigner = threeIdProvider.keychain._keyring.getSigner();
+
+    // 2. Create tile
+    const tile = await TileDocument.create(ceramicClient, {
+        stage: "Signed by vanilla",
+    });
+    await tile.update({ stage: "Signed second time" }, undefined, {
+        anchor: true,
+    });
+    await waitForAnchor(tile, 60*60);
```

Comment:
this means waiting an hour before the test fails if the anchor batch fails, which seems very long.  The other tests use 15 minutes, probably makes sense to be consistent with them.  Might even be nice to pull ANCHOR_TIMEOUT to be a re-usable constant that the tests share

Reply:
done